### PR TITLE
hbit-herc20 follow up, remove usage of rfc003

### PR DESCRIPTION
**Pull request by tcharding**
This builds on top of @luckysori' and @da-kami's  awesome work adding hbit, super stoked with that PR :)

First commit is from: https://github.com/comit-network/comit-rs/pull/2776

Removes usage of rfc003 types from the newly added hbit code. We do this be removing some traits and defining methods directly on `SwapSeed` and `RootSeed` as needed. Also do a bunch of re-naming stuff with a prefix 'rfc003' pre-pended in order not to have name clashes.

### CODE REVIEW

**Comment by tcharding** [comment]
Totes should have used the [PR retargeting](https://github.blog/changelog/2020-05-19-pull-request-retargeting/) that you told me about @thomaseizinger!

**Comment by luckysori** [comment]
Once you rebase I'll take a look :stuck_out_tongue: Removing myself from reviewers so that you can request later.

File: cnd/src/seed.rs [hunk 1]
```diff
@@ -191,6 +192,18 @@ impl SwapSeed {
     pub fn derive_secret(&self) -> Secret {
         self.sha256_with_seed(&[b"SECRET"]).into()
     }
+
+    /// Used to derive the transient redeem identity for hbit swaps.
+    pub fn derive_transient_redeem_identity(&self) -> SecretKey {
```
Comment (luckysori) [inline cnd/src/seed.rs:197]: :question: Shouldn't these functions live in a `hbit` module? It seems to me like the swap seed shouldn't know about protocols.

Comment (thomaseizinger) [reply cnd/src/seed.rs:197]: Good call, I like that idea :)

**Pull request merged**